Makefile

```diff
@@ -217,12 +219,14 @@ install:
 lint:
 	GO111MODULE=on golangci-lint run -v --timeout=10m
 
-# We have to run test twice: once for all packages with -race and then once more without -race
-# for packages that have known race detection issues
-# Run tests with -online flag set to true, to include tests requiring network connectivity in CI
+# We have to run test twice: once for all packages with -race and then once
+# more without -race for packages that have known race detection issues.
+#
+# Run tests with -tags=has_network to include tests that require network
+# connectivity.
 test:
-	CGO_ENABLED=1 go test $(CGO_FLAGS) -race -cover -coverprofile=cover.out -p=4 ./... -- -online
-	CGO_ENABLED=1 go test $(CGO_FLAGS) -cover -coverprofile=cover-norace.out -p=4 ./pkg/integrations/node_exporter ./pkg/logs -- -online
```

Comment:
Turns out flags after `--` aren't read and you can't omit the `--` otherwise `-online` will be passed to every package, even packages that don't have them defined. Build tags are the best choice here.